nsv13/code/modules/power/rbmk.dm

```diff
@@ -0,0 +1,864 @@
+		var/heat_delta = (KELVIN_TO_CELSIUS(coolant_input.return_temperature()) / 100) * gas_absorption_effectiveness //Take in the gas as a cooled input, cool the reactor a bit. The optimum, 100% balanced reaction sits at K=1, coolant input temp of 200K / -73 celsius.
+		last_heat_delta = heat_delta
+		temperature += heat_delta
+		coolant_output.merge(coolant_input) //And now, shove the input into the output.
+		coolant_input.clear() //Clear out anything left in the input gate.
+		color = null
+	else
+		if(has_fuel())
+			temperature += temperature / 500 //This isn't really harmful early game, but when your reactor is up to full power, this can get out of hand quite quickly.
+			vessel_integrity -= temperature / 200 //Think fast loser.
+			take_damage(10) //Just for the sound effect, to let you know you've fucked up.
+			color = "[COLOR_RED]"
+	//Now, heat up the output and set our pressure.
+	coolant_output.set_temperature(CELSIUS_TO_KELVIN(temperature)) //Heat the coolant output gas that we just had pass through us.
+	last_output_temperature = KELVIN_TO_CELSIUS(coolant_output.return_temperature())
+	pressure = KPA_TO_PSI(coolant_output.return_pressure())
+	power = (temperature / RBMK_TEMPERATURE_CRITICAL) * 100
+	var/radioactivity_spice_multiplier = 1 //Some gasses make the reactor a bit spicy.
+	var/depletion_modifier = 0.035 //How rapidly do your rods decay
+	gas_absorption_effectiveness = gas_absorption_constant
+	//Next up, handle moderators!
+	if(moderator_input.total_moles() >= minimum_coolant_level)
+		var/total_fuel_moles = moderator_input.get_moles(/datum/gas/plasma) + (moderator_input.get_moles(/datum/gas/constricted_plasma)*2)+ (moderator_input.get_moles(/datum/gas/tritium)*10) //Constricted plasma is 50% more efficient as fuel than plasma, but is harder to produce
+		var/power_modifier = max((moderator_input.get_moles(/datum/gas/oxygen) / moderator_input.total_moles() * 10), 1) //You can never have negative IPM. For now.
+		if(total_fuel_moles >= minimum_coolant_level) //You at least need SOME fuel.
+			var/power_produced = max((total_fuel_moles / moderator_input.total_moles() * 10), 1)
+			last_power_produced = max(0,((power_produced*power_modifier)*moderator_input.total_moles()))
+			last_power_produced *= (power/100) //Aaaand here comes the cap. Hotter reactor => more power.
+			last_power_produced *= base_power_modifier //Finally, we turn it into actual usable numbers.
+			radioactivity_spice_multiplier += moderator_input.get_moles(/datum/gas/tritium) / 5 //Chernobyl 2.
+			var/turf/T = get_turf(src)
+			if(power >= 20)
+				coolant_output.adjust_moles(/datum/gas/nucleium, total_fuel_moles/50) //Shove out nucleium into the air when it's fuelled. You need to filter this off, or you're gonna have a bad time.
+			var/obj/structure/cable/C = T.get_cable_node()
+			if(!C || !C.powernet)
+				return
+			else
+				C.powernet.newavail += last_power_produced
```

Comment:
might want to check how TEG passes power to powernet to reduce bugs.

Reply:
Agreed. This feels wrong and hacky.

Reply:
Okay, this is TECHNICALLY correct, but he shouldn't be doing it directly.
He should be using `/obj/machinery/power/proc/add_avail(amount)`